File: CSharp/LeetCode.Test/003-LongestSubstringWithoutRepeatingCharacters-Test.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace LeetCode.Test
{
    [TestClass]
    public class _003_LongestSubstringWithoutRepeatingCharacters_Test
    {
        [TestMethod]
        public void GeneralTest()
        {
            var input = "abcabcbb";

            var solution = new _003_LongestSubstringWithoutRepeatingCharacters();
            var result = solution.LengthOfLongestSubstring(input);

            Assert.AreEqual(3, result);
        }

        [TestMethod]
        public void RepeatCharsTest()
        {
            var input = "bbbbb";

            var solution = new _003_LongestSubstringWithoutRepeatingCharacters();
            var result = solution.LengthOfLongestSubstring(input);

            Assert.AreEqual(1, result);
        }

        [TestMethod]
        public void EmptyInputTest()
        {
            var solution = new _003_LongestSubstringWithoutRepeatingCharacters();
            var result = solution.LengthOfLongestSubstring("");

            Assert.AreEqual(0, result);
        }

        [TestMethod]
        public void LongStringTest()
        {
            var input = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz";

            var solution = new _003_LongestSubstringWithoutRepeatingCharacters();
            var result = solution.LengthOfLongestSubstring(input);

            Assert.AreEqual(26, result);
        }

        [TestMethod]
        public void LongStringTest_HaveShortRepeat()
        {
            var input = "abcdefghijklmnopqrstuvwxyzabcdefghijk  lmnopqrstuvwxyz";

            var solution = new _003_LongestSubstringWithoutRepeatingCharacters();
            var result = solution.LengthOfLongestSubstring(input);

            Assert.AreEqual(27, result);
        }
    }
}

File: C++/LeetCode.Test/003-LongestSubstringWithoutRepeatingCharacters-Test.cpp
#include "stdafx.h"
#include "CppUnitTest.h"
#include "..\LeetCode\003-LongestSubstringWithoutRepeatingCharacters.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace LeetCode;

namespace LeetCodeTest
{
    TEST_CLASS(_003_LongestSubstringWithoutRepeatingCharacters_Test)
    {
    public:

        TEST_METHOD(LengthOfLongestSubstringTest_General)
        {
            string input = "abcabcbb";

            _003_LongestSubstringWithoutRepeatingCharacters solution;
            int result = solution.lengthOfLongestSubstring(input);

            Assert::AreEqual(3, result);
        }

        TEST_METHOD(LengthOfLongestSubstringTest_RepeatChars)
        {
            string input = "bbbbb";

            _003_LongestSubstringWithoutRepeatingCharacters solution;
            int result = solution.lengthOfLongestSubstring(input);

            Assert::AreEqual(1, result);
        }

        TEST_METHOD(LengthOfLongestSubstringTest_EmptyInput)
        {
            string input = "";

            _003_LongestSubstringWithoutRepeatingCharacters solution;
            int result = solution.lengthOfLongestSubstring(input);

            Assert::AreEqual(0, result);
        }

        TEST_METHOD(LengthOfLongestSubstringTest_LongString)
        {
            string input = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz";

            _003_LongestSubstringWithoutRepeatingCharacters solution;
            int result = solution.lengthOfLongestSubstring(input);

            Assert::AreEqual(26, result);
        }

        TEST_METHOD(LengthOfLongestSubstringTest_LongString_HaveShortRepeat)
        {
            string input = "abcdefghijklmnopqrstuvwxyzabcdefghijk  lmnopqrstuvwxyz";

            _003_LongestSubstringWithoutRepeatingCharacters solution;
            int result = solution.lengthOfLongestSubstring(input);

            Assert::AreEqual(27, result);
        }

    };
}